src/test/java/ca/qaguru/oranghrmbatch27/pages/JobCategoriesPage.java

```diff
@@ -0,0 +1,83 @@
+package ca.qaguru.oranghrmbatch27.pages;
+
+import ca.qaguru.oranghrmbatch27.library.PageBase;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.testng.Assert;
+
+import java.util.List;
+
+public class JobCategoriesPage extends PageBase {
+    WebDriver driver;
+    private final String addJobCategoryBtn = "//button[text()=' Add ']";
+    private final String categoryNameTxtField = "//label[text()='Name']/following::input";
+    private final String categorySaveBtn = "//button[text()=' Save ']";
+
+    private final String categoryCancelBtn = "//button[text()=' Cancel ']";
+
+    private final String categoryExistMsg = "//div[@class='oxd-form-row']/div";
+    private final String categoryHeader = "//h6[text()='Job Categories']";
+    private final String addJobCategoryHeader = "//h6[text()='Add Job Category']";
+    private final String listOfJobCategories = ".oxd-table-body";
+    private final String categories = "//div[@class='oxd-table-body']/div[@class='oxd-table-card']";
+    @FindBy(xpath = categories)
+    private List<WebElement> listCategory;
+
+    public JobCategoriesPage(WebDriver driver) {
+        super(driver);
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+
+    }
+
+    public void header() {
```

Comment:
generic method can be created
Assertions should be on Test level of testng- JobcategoryTests

Reply:
updated